contributors/design-proposals/daemonset-update.md

```diff
@@ -129,18 +127,35 @@ type DaemonSetStatus struct {
 
 	// UpdatedNumberScheduled is the total number of nodes that are running updated
 	// daemon pod
+	// +optional
 	UpdatedNumberScheduled int32 `json:"updatedNumberScheduled"`
 
 	// NumberAvailable is the number of nodes that should be running the
 	// daemon pod and have one or more of the daemon pod running and
 	// available (ready for at least minReadySeconds)
+	// +optional
 	NumberAvailable int32 `json:"numberAvailable"`
 
 	// NumberUnavailable is the number of nodes that should be running the
 	// daemon pod and have non of the daemon pod running and available
 	// (ready for at least minReadySeconds)
+	// +optional
 	NumberUnavailable int32 `json:"numberUnavailable"`
 }
+
+type DaemonSet struct {
+	// A sequence number representing a specific generation of the template.
+	// Populated by the system. Read-only.
+	// +optional
+	TemplateGeneration int64 `json:"templateGeneration,omitempty"`
+}
+
+const (
+	// DaemonSetTemplateGenerationKey is the key of the labels that is added
+	// to daemon set pods to distinguish between old and new pod templates
+	// during DaemonSet template update.
+	DaemonSetTemplateGenerationKey string = "pod-template-generation"
```

Comment:
See Brian's previous comment:
> The desire for lack of a namespace prefix was one reason I had originally insisted on the key being configurable -- so that the user could change the key in order to avoid key-name collisions with their own labels.
However, in the short term it's easier if we don't support that field. If users request it, we can add it back later.

We should instead make the label key configurable to avoid collisions. 

Because we want the label key to be non-namespaced, we originally designed Deployment "pod template hash" label key to be configurable for users to avoid key name collisions. We haven't, however, implement that in Deployment, since we haven't see no user request for that yet.